docs/devsite-help/troubleshooting.md

```diff
@@ -218,3 +218,62 @@ native libraries are copied appropriately.
 
 This information is now in the [protocol buffers guide](protobuf.md#repeated-fields-and-maps); this section has been
 retained for the sake of existing links.
+
+## How can I get ErrorInfo from RpcException?
+
+`ErrorInfo` in `RpcException` provides a consistent and human readable error details.
+It contains three major fields - reason, domain and metadata described as below:
+
+- `Reason` - It is a constant value and indicates the proximate cause of 
+the error. Error reasons are unique within a particular domain of errors.
+
+- `Domain` - It is the logical grouping to which the error belongs. It is 
+typically the registered service name of the tool or product that 
+generates the error. Example: "pubsub.googleapis.com". If the error 
+is generated by some common infrastructure, the error domain must be 
+a globally unique value that identifies the infrastructure. 
+For Google API infrastructure, the error domain is "googleapis.com".
+
+- `Metadata`- It is the additional structured details about the error. 
+It can help in identifying the exact cause of the error.
+
+```csharp
+using Google.Api.Gax.Grpc;
+try
+{
+  // Code that can throw the RpcException.
+}
+catch (RpcException  ex)
+{
+   var errorInfo = ex.GetErrorInfo();   
+   var domain = errorInfo.Domain;
+   var reason = errorInfo.Reason;
+   var metadata = errorInfo.Metadata;
+
+   // Use the domain, reason and metadata of ErrorInfo, e.g. for logging or taking remedial action.
+}
+```
+
+### How can I get RpcException from SpannerException?
+
+The `RpcException` property of `SpannerException` returns the underlying
+`RpcException` which caused the `SpannerException`. The returned 
+value may be null if the `SpannerException` was not caused by `RpcException`.
+
+`RpcException` property of `SpannerException` can be used like:
+
+```csharp
+using Google.Api.Gax.Grpc;
+try
+{
+  // Code using Spanner 
+}
+catch (SpannerException ex)
+{
+   var rpcException = ex.RpcException;   
+   if(rpcException != null)
```

Comment:
missing space between `if` and `(`
```suggestion
   if (rpcException != null)
```

Reply:
Also, I see in https://github.com/googleapis/google-cloud-dotnet/pull/9484#pullrequestreview-1234636109 that Jon mentions that
```
I would definitely suggest changing this to be focused on RpcException. We can have one example showing that in
detail. We can have one example showing that in detail... then a subsection for SpannerException, with a brief
example using the pattern match (and maybe having another catch clause for SpannerException when there isn't an
inner RpcException?) that refers to the earlier example.
```
so note on `SpannerException`  should be a subsection of the above `RpcException` section instead of a new section (I may be wrong, but wanted to check and be sure)